bootloaders/eboot/eboot_command.c

```diff
@@ -28,38 +29,248 @@ uint32_t eboot_command_calculate_crc32(const struct eboot_command* cmd)
                       offsetof(struct eboot_command, crc32));
 }
 
-int eboot_command_read(struct eboot_command* cmd)
+#if defined (EBOOT_ENABLE_FLASH_STORAGE)
+
+uint32_t eboot_read_flash_index(eboot_index_t *eboot_index)
+{
+    if (SPIRead(0 + APP_START_OFFSET + sizeof(image_header_t) + sizeof(section_header_t), &eboot_index, sizeof(eboot_index))) {
+        return 0;
+    }
+
+    if (eboot_index->magic != EBOOT_INDEX_MAGIC_V1) {
+        return 0;
+    }
+
+    return 1;
+}
+
+eboot_flash_command_t *commandAddress(void) {
+  eboot_index_t eboot_index;
+  const uint32_t addr = 0 + APP_START_OFFSET + sizeof(image_header_t) + sizeof(section_header_t);
+  if (SPIRead(addr, (uint32_t *)&eboot_index, sizeof(eboot_index))) {
+    return NULL;
+  }
+  if (eboot_index.magic != EBOOT_INDEX_MAGIC_V1) {
+    return NULL;
+  } else {
+    return eboot_index.commands;
+  }
+}
+
+bool readBootCommand(int cmd, eboot_flash_command_t *dst) {
+  eboot_flash_command_t *cmds = commandAddress();
+  if (cmd >= EBOOT_COMMAND_MAX_COUNT) {
+    return 0;
+  }
+#if defined (EBOOT_ENABLE_SERIAL_DEBUG)
+    ets_putc('r');
+    ets_putc('b');
+    ets_putc('0' + cmd);
+    ets_putc('\n');
+#endif
+  uint32_t uint = (uint32_t) cmds;
+  uint32_t addr = (uint32_t)cmds - 0x40200000;
```

Comment:
Maybe use a define instead of hardcoded number, to give the reader the idea where it's coming from